### DISCUSSION TITLE
Potential UX enhancements to V4 filters web part

### DISCUSSION THREAD

**Discussion by mbice** [microsoft-search/pnp-modern-search#1707] [Ideas]
Hi,

Figured this might be the best place to discuss a handful of UX questions I have about the Filters web part changes in V4. Scenario I'm using as an example for everything below is multi-select taxonomy filters.

### Displaying the AND/OR toggle to users

I see the value in having the AND/OR toggle for power users, but the feedback we are receiving is that general end-users don't really understand what this is and they are unsure which one is selected (the selected value looks like a clickable hyperlink right now). In most usage scenarios we would prefer to enforce an operator and not have it changeable by the end user. Does it make sense to have this be a configurable property on the web part (e.g. checkbox to enable user switching operators)?

### Scrolling when > 10 items

What would your thoughts be on having a threshold in the web part properties that could be set to determine when the scrollbars show up? We are already seeing scenarios where there is enough space in the column to show more options, but the scrolling kicks in anyway and forces users to scroll down. Would be nice to have more visible at a glance for situations where we know there may only be 12-15 items in the refiner.

### Search box

Somewhat related to the scrolling threshold point above.. I see two issues with the "Search a value" filter search. First, it's not styled in a way that is obvious it's a search box (it looks more like a heading with a border underneath it). Second, when have few enough items that scrolling is not enabled, you don't really need the search and it created a lot of unnecessary visual distraction. Would this also make sense to have an enabled/disabled toggle for it OR have a threshold property to define how many items have to be in the refiner for it to show up?

### Selected tags

The selected tags seem redundant when you have a few number of filters available. They are clearly marked as selected by the checkbox next to them and it most of the usage scenarios we are implementing (refiners with a dozen or less values) having the selected tags sit immediately above the checked checkboxes seems like visual noise and redundant.

Thank you for taking the time to read through all of this. Thoughts on any of the above suggestions?

Cheers

**Comment by DavorSukeri** [comment]
I definitely agree to all your suggestions and I might add few more:

- Checkboxes have too large space between them. If the space would be 30-50% less, it would still be clear to read and most importantly it would be more compact. In my opinion it takes too much space now.

- Filter display name is also too large (18px) in my opinion and it kind of pops out from the rest of the Sharepoint page and looks like it doesn't belong there. Lets say make the font size 16px so it is still bigger than the actual filtervalue font which is 14px, but not unneccessary large.

**Reply by kasperbolarsen** [reply]
Reducing the size of the Checkboxes is easy, change
	.filter--value {
			white-space: nowrap;
			overflow: hidden;
			text-overflow: ellipsis;
			display: flex;
			align-items: center;
		}
to 
	.filter--value {
			white-space: nowrap;
			overflow: hidden;
			text-overflow: ellipsis;
			display: flex;
			align-items: center;
            
            max-height: 25px;
		}
and it looks way more compact, to compact for my taste ;-)
![image](https://github.com/user-attachments/assets/37974cac-ed0b-4c67-8921-68172f556dba)

Comparing the size of the Filter display name to the Web part title I think it is a fair compromise, but perhaps you are not using the Web part title ?

**Reply by DavorSukeri** [reply]
Thanks for the tip!

Actually no I’m using webpart titles at all in my scenario where I have built custom search results page to replace OOTB search results page.

Titles are used only in search verticals but not in Search box, Filter or Search results -webparts. Search box and Filters are self-explanatory so they dont need webpart title and Search results webparts neither because the search vertical already says which results I am looking at.

Without webpart titles the page looks more clean and tidy. :)

**Reply by DavorSukeri** [reply]
@kasperbolarsen that checkbox size reduction works great except that it also affects the Date Range filter so that the date picker does not show fully. And yes the filter display name size doesn't look that bad if web part title is used. :)

**Comment by wobba** [comment]
I like all the suggestions and look forward to a PR :)
